### PR TITLE
Replace calls to .Any() with .Count is not 0.

### LINKED ISSUE #36: Replace calls to ``.Any()`` with ``.Count is not 0``.
# Issue Details

**Is your optimization request related to a problem? Please describe.**
Currently the codebase uses ``.Any()`` inside of DisposableCodeWriter.cs 6 times per method inside of it. This can result in performance issues inside of the Visual Studio IDE and should be corrected.

**Describe the solution you'd like**
Replace ``.Any()`` with ``.Count is not 0`` to make the code overall faster.

**Describe alternatives you've considered**
No other alternatives are needed to be considered in this case as this is a simple optimization.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by AraHaan**
# Checklist

- [x] I have read the Code of Conduct and the Contributing files before opening this issue.
- [x] I have verified the issue this pull request fixes or feature this pull request implements is to the current release.
- [x] I am using the latest stable release with the above code fix or the current main branch if code changes are present (prerelease code changes).
- [x] I have debugged and tested my changes before submitting this issue and that everything should just work.

## Describe the issue this fixes / feature this adds

Fixes #36.

### CODE REVIEW

File: src/IDisposableGenerator/WorkItemCollection.cs [hunk 1]
```diff
@@ -1,20 +1,17 @@
 namespace IDisposableGenerator;
 
-internal class WorkItemCollection
+internal class WorkItemCollection(Compilation compilation)
```
[the remark this answers is not in the document]
Comment (AraHaan) [reply src/IDisposableGenerator/WorkItemCollection.cs:3]: 😄 When it thinks this is empty.

**Review by Rekkonnect — APPROVED** [review]
Minor nits

File: src/IDisposableGenerator/WorkItemCollection.cs [hunk 2]
```diff
@@ -28,7 +25,7 @@ public void Process(INamedTypeSymbol testClass, CancellationToken ct)
         var classItemsQuery =
             from att in testClass.GetAttributes()
             where att.AttributeClass!.Name.Equals(
-                "GenerateDisposeAttribute")
+                "GenerateDisposeAttribute", StringComparison.Ordinal)
```
[the remark this answers is not in the document]
Comment (AraHaan) [reply src/IDisposableGenerator/WorkItemCollection.cs:28]: I would like that very much however the attribute code files does not get directly compiled to be a part of the generator assembly, instead it gets provided as basically an output from the generator itself. I guess a workaround is to reference the generator and having it run here and then use ``nameof(GenerateDisposeAttribute)`` though.

[the remark this answers is not in the document]
Comment (Rekkonnect) [reply src/IDisposableGenerator/WorkItemCollection.cs:28]: If it's included in the generator, it's fine. I don't find a reason to change how the API for the generator is accessible to the user.

**Pull request merged**